File: resour/sources/com/yhchat/canarys/p007ui/components/ImageUtils.java
package com.yhchat.canarys.p007ui.components;

import android.content.Context;
import android.content.SharedPreferences;
import android.os.Build;
import coil.ComponentRegistry;
import coil.ImageLoader;
import coil.decode.GifDecoder;
import coil.decode.ImageDecoderDecoder;
import coil.decode.SvgDecoder;
import coil.request.ImageRequest;
import kotlin.Metadata;
import kotlin.jvm.internal.Intrinsics;
import kotlin.text.StringsKt;

/* compiled from: ImageUtils.kt */
@Metadata(m168d1 = {"\u0000,\n\u0002\u0018\u0002\n\u0002\u0010\u0000\n\u0002\b\u0003\n\u0002\u0010\u000b\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0010\u000e\n\u0002\b\u0006\b\u00c7\u0002\u0018\u00002\u00020\u0001B\t\b\u0002\u00a2\u0006\u0004\b\u0002\u0010\u0003J\u0010\u0010\u0004\u001a\u00020\u00052\u0006\u0010\u0006\u001a\u00020\u0007H\u0002J\u000e\u0010\b\u001a\u00020\t2\u0006\u0010\u0006\u001a\u00020\u0007J \u0010\n\u001a\u00020\u000b2\u0006\u0010\u0006\u001a\u00020\u00072\u0006\u0010\f\u001a\u00020\r2\b\b\u0002\u0010\u000e\u001a\u00020\u0005J \u0010\u000f\u001a\u00020\u000b2\u0006\u0010\u0006\u001a\u00020\u00072\u0006\u0010\f\u001a\u00020\r2\b\b\u0002\u0010\u000e\u001a\u00020\u0005J \u0010\u0010\u001a\u00020\u000b2\u0006\u0010\u0006\u001a\u00020\u00072\u0006\u0010\f\u001a\u00020\r2\b\b\u0002\u0010\u000e\u001a\u00020\u0005J \u0010\u0011\u001a\u00020\u000b2\u0006\u0010\u0006\u001a\u00020\u00072\u0006\u0010\f\u001a\u00020\r2\b\b\u0002\u0010\u000e\u001a\u00020\u0005J \u0010\u0012\u001a\u00020\u000b2\u0006\u0010\u0006\u001a\u00020\u00072\u0006\u0010\f\u001a\u00020\r2\b\b\u0002\u0010\u000e\u001a\u00020\u0005\u00a8\u0006\u0013"}, m169d2 = {"Lcom/yhchat/canarys/ui/components/ImageUtils;", "", "<init>", "()V", "isDataSaverEnabled", "", "context", "Landroid/content/Context;", "createImageLoader", "Lcoil/ImageLoader;", "createStickerImageRequest", "Lcoil/request/ImageRequest;", "url", "", "enableHardware", "createImageRequest", "createAvatarImageRequest", "createBotImageRequest", "createBoardImageRequest", "app_debug"}, m170k = 1, m171mv = {2, 2, 0}, m173xi = 48)
/* loaded from: classes5.dex */
public final class ImageUtils {
    public static final int $stable = 0;
    public static final ImageUtils INSTANCE = new ImageUtils();

    private ImageUtils() {
    }

    private final boolean isDataSaverEnabled(Context context) {
        SharedPreferences prefs = context.getSharedPreferences("display_settings", 0);
        return prefs.getBoolean("data_saver", false);
    }

    public final ImageLoader createImageLoader(Context context) {
        Intrinsics.checkNotNullParameter(context, "context");
        ImageLoader.Builder builder = new ImageLoader.Builder(context);
        ComponentRegistry.Builder builder2 = new ComponentRegistry.Builder();
        builder2.add(new GifDecoder.Factory(false, 1, null));
        if (Build.VERSION.SDK_INT >= 28) {
            builder2.add(new ImageDecoderDecoder.Factory(false, 1, null));
        }
        builder2.add(new SvgDecoder.Factory(false, 1, null));
        return builder.components(builder2.build()).build();
    }

    public static /* synthetic */ ImageRequest createStickerImageRequest$default(ImageUtils imageUtils, Context context, String str, boolean z, int i, Object obj) {
        if ((i & 4) != 0) {
            z = true;
        }
        return imageUtils.createStickerImageRequest(context, str, z);
    }

    public final ImageRequest createStickerImageRequest(Context context, String url, boolean enableHardware) {
        Intrinsics.checkNotNullParameter(context, "context");
        Intrinsics.checkNotNullParameter(url, "url");
        if (isDataSaverEnabled(context)) {
            return new ImageRequest.Builder(context).data(null).build();
        }
        ImageRequest.Builder builder = new ImageRequest.Builder(context).data(url).allowHardware(enableHardware).crossfade(true);
        if (StringsKt.contains$default((CharSequence) url, (CharSequence) "chat-img.jwznb.com", false, 2, (Object) null) || StringsKt.contains$default((CharSequence) url, (CharSequence) "jwznb.com", false, 2, (Object) null) || StringsKt.contains$default((CharSequence) url, (CharSequence) "myapp.jwznb.com", false, 2, (Object) null)) {
            builder.setHeader("Referer", "https://myapp.jwznb.com");
            builder.setHeader("User-Agent", "Mozilla/5.0 (Linux; Android 10; Mobile) AppleWebKit/537.36");
        }
        return builder.build();
    }

    public static /* synthetic */ ImageRequest createImageRequest$default(ImageUtils imageUtils, Context context, String str, boolean z, int i, Object obj) {
        if ((i & 4) != 0) {
            z = true;
        }
        return imageUtils.createImageRequest(context, str, z);
    }

    public final ImageRequest createImageRequest(Context context, String url, boolean enableHardware) {
        Intrinsics.checkNotNullParameter(context, "context");
        Intrinsics.checkNotNullParameter(url, "url");
        if (isDataSaverEnabled(context)) {
            return new ImageRequest.Builder(context).data(null).build();
        }
        ImageRequest.Builder builder = new ImageRequest.Builder(context).data(url).allowHardware(enableHardware).crossfade(true);
        if (StringsKt.contains$default((CharSequence) url, (CharSequence) "chat-img.jwznb.com", false, 2, (Object) null) || StringsKt.contains$default((CharSequence) url, (CharSequence) "jwznb.com", false, 2, (Object) null) || StringsKt.contains$default((CharSequence) url, (CharSequence) "myapp.jwznb.com", false, 2, (Object) null)) {
            builder.setHeader("Referer", "https://myapp.jwznb.com");
            builder.setHeader("User-Agent", "Mozilla/5.0 (Linux; Android 10; Mobile) AppleWebKit/537.36");
        }
        return builder.build();
    }

    public static /* synthetic */ ImageRequest createAvatarImageRequest$default(ImageUtils imageUtils, Context context, String str, boolean z, int i, Object obj) {
        if ((i & 4) != 0) {
            z = true;
        }
        return imageUtils.createAvatarImageRequest(context, str, z);
    }

    public final ImageRequest createAvatarImageRequest(Context context, String url, boolean enableHardware) {
        Intrinsics.checkNotNullParameter(context, "context");
        Intrinsics.checkNotNullParameter(url, "url");
        if (isDataSaverEnabled(context)) {
            return new ImageRequest.Builder(context).data(null).build();
        }
        ImageRequest.Builder builder = new ImageRequest.Builder(context).data(url).allowHardware(enableHardware).crossfade(true);
        if (StringsKt.contains$default((CharSequence) url, (CharSequence) "chat-img.jwznb.com", false, 2, (Object) null) || StringsKt.contains$default((CharSequence) url, (CharSequence) "jwznb.com", false, 2, (Object) null) || StringsKt.contains$default((CharSequence) url, (CharSequence) "myapp.jwznb.com", false, 2, (Object) null)) {
            builder.setHeader("Referer", "https://myapp.jwznb.com");
            builder.setHeader("User-Agent", "Mozilla/5.0 (Linux; Android 10; Mobile) AppleWebKit/537.36");
        }
        return builder.build();
    }

    public static /* synthetic */ ImageRequest createBotImageRequest$default(ImageUtils imageUtils, Context context, String str, boolean z, int i, Object obj) {
        if ((i & 4) != 0) {
            z = true;
        }
        return imageUtils.createBotImageRequest(context, str, z);
    }

    public final ImageRequest createBotImageRequest(Context context, String url, boolean enableHardware) {
        Intrinsics.checkNotNullParameter(context, "context");
        Intrinsics.checkNotNullParameter(url, "url");
        if (isDataSaverEnabled(context)) {
            return new ImageRequest.Builder(context).data(null).build();
        }
        return new ImageRequest.Builder(context).data(url).setHeader("Referer", "https://myapp.jwznb.com").setHeader("User-Agent", "Mozilla/5.0 (Linux; Android 10; Mobile) AppleWebKit/537.36").allowHardware(enableHardware).crossfade(true).build();
    }

    public static /* synthetic */ ImageRequest createBoardImageRequest$default(ImageUtils imageUtils, Context context, String str, boolean z, int i, Object obj) {
        if ((i & 4) != 0) {
            z = true;
        }
        return imageUtils.createBoardImageRequest(context, str, z);
    }

    public final ImageRequest createBoardImageRequest(Context context, String url, boolean enableHardware) {
        Intrinsics.checkNotNullParameter(context, "context");
        Intrinsics.checkNotNullParameter(url, "url");
        if (isDataSaverEnabled(context)) {
            return new ImageRequest.Builder(context).data(null).build();
        }
        return new ImageRequest.Builder(context).data(url).setHeader("Referer", "https://myapp.jwznb.com").setHeader("User-Agent", "Mozilla/5.0 (Linux; Android 10; Mobile) AppleWebKit/537.36").allowHardware(enableHardware).crossfade(true).build();
    }
}

File: app/src/main/java/com/yhchat/canary/ui/components/ImageUtils.kt
package com.yhchat.canary.ui.components

import android.content.Context
import coil.ImageLoader
import coil.decode.GifDecoder
import coil.decode.ImageDecoderDecoder
import coil.decode.SvgDecoder
import coil.request.CachePolicy
import coil.request.ImageRequest
import coil.size.Size
import coil.size.Scale

/**
 * 图片加载工具类
 * 支持多种图片格式包括 GIF、AVIF、WEBP、SVG 等
 * 内存优化: 支持图片尺寸采样、缓存策略、RGB565格式
 */
object ImageUtils {
    
    // 预定义尺寸常量 - 用于统一管理图片加载尺寸
    object ImageSize {
        const val AVATAR_SMALL = 48   // 小头像 (会话列表)
        const val AVATAR_MEDIUM = 72  // 中头像 (聊天消息)
        const val AVATAR_LARGE = 120  // 大头像 (个人资料)
        const val STICKER_SMALL = 64  // 小表情
        const val STICKER_MEDIUM = 96 // 中表情
        const val THUMBNAIL = 280     // 缩略图 (消息列表中的图片)
        const val PREVIEW = 720       // 预览图 (图片查看器初始加载)
    }
    
    private fun isDataSaverEnabled(context: Context): Boolean {
        val prefs = context.getSharedPreferences("display_settings", Context.MODE_PRIVATE)
        return prefs.getBoolean("data_saver", false)
    }
    
    /**
     * 为URL添加七牛云图片处理参数，实现服务端缩略图
     * @param url 原始图片URL
     * @param width 目标宽度
     * @param height 目标高度 (可选，不传则等比缩放)
     */
    fun appendQiniuThumbnail(url: String?, width: Int, height: Int? = null): String? {
        if (url.isNullOrBlank()) return url
        if (url.contains("?")) return url // 已有参数，不再添加
        
        return if (height != null) {
            "$url?imageView2/2/w/$width/h/$height"
        } else {
            "$url?imageView2/2/w/$width"
        }
    }
    
    /**
     * 创建支持多格式的ImageLoader（已废弃，建议使用MainActivity中的全局ImageLoader）
     */
    @Deprecated("使用 Coil.imageLoader(context) 获取全局优化的ImageLoader")
    fun createImageLoader(context: Context): ImageLoader {
        return ImageLoader.Builder(context)
            .components {
                add(GifDecoder.Factory())
                if (android.os.Build.VERSION.SDK_INT >= 28) {
                    add(ImageDecoderDecoder.Factory())
                }
                add(SvgDecoder.Factory())
            }
            .allowRgb565(true)
            .build()
    }
    
    /**
     * 创建表情图片请求 - 内存优化版
     * 限制表情图片加载尺寸，减少内存占用
     */
    fun createStickerImageRequest(
        context: Context,
        url: String,
        enableHardware: Boolean = true,
        size: Int = ImageSize.STICKER_MEDIUM
    ): ImageRequest {
        if (isDataSaverEnabled(context)) {
            return ImageRequest.Builder(context)
                .data(null)
                .build()
        }
        val builder = ImageRequest.Builder(context)
            .data(url)
            .allowHardware(enableHardware)
            .crossfade(true)
            .size(size) // 限制加载尺寸
            .scale(Scale.FIT)
            .memoryCachePolicy(CachePolicy.ENABLED)
            .diskCachePolicy(CachePolicy.ENABLED)
            
        // 为相关域名添加Referer头
        if (url.contains("chat-img.jwznb.com") || url.contains("jwznb.com") || url.contains("myapp.jwznb.com")) {
            builder.setHeader("Referer", "https://myapp.jwznb.com")
            builder.setHeader("User-Agent", "Mozilla/5.0 (Linux; Android 10; Mobile) AppleWebKit/537.36")
        }
        
        return builder.build()
    }
    
    /**
     * 创建普通图片请求 - 内存优化版
     * @param maxSize 最大尺寸限制，0表示不限制
     */
    fun createImageRequest(
        context: Context,
        url: String,
        enableHardware: Boolean = true,
        maxSize: Int = ImageSize.THUMBNAIL
    ): ImageRequest {
        if (isDataSaverEnabled(context)) {
            return ImageRequest.Builder(context)
                .data(null)
                .build()
        }
        val builder = ImageRequest.Builder(context)
            .data(url)
            .allowHardware(enableHardware)
            .crossfade(true)
            .memoryCachePolicy(CachePolicy.ENABLED)
            .diskCachePolicy(CachePolicy.ENABLED)
        
        // 设置最大尺寸限制
        if (maxSize > 0) {
            builder.size(maxSize)
            builder.scale(Scale.FIT)
        }
            
        // 为相关域名添加Referer头
        if (url.contains("chat-img.jwznb.com") || url.contains("jwznb.com") || url.contains("myapp.jwznb.com")) {
            builder.setHeader("Referer", "https://myapp.jwznb.com")
            builder.setHeader("User-Agent", "Mozilla/5.0 (Linux; Android 10; Mobile) AppleWebKit/537.36")
        }
        
        return builder.build()
    }
    
    /**
     * 创建头像图片请求 - 内存优化版
     * 头像固定较小尺寸，大幅减少内存占用
     * @param avatarSize 头像尺寸，默认中等头像
     */
    fun createAvatarImageRequest(
        context: Context,
        url: String,
        enableHardware: Boolean = true,
        avatarSize: Int = ImageSize.AVATAR_MEDIUM
    ): ImageRequest {
        if (isDataSaverEnabled(context)) {
            return ImageRequest.Builder(context)
                .data(null)
                .build()
        }
        
        // 尝试使用七牛云服务端缩略图
        val optimizedUrl = appendQiniuThumbnail(url, avatarSize, avatarSize) ?: url
        
        val builder = ImageRequest.Builder(context)
            .data(optimizedUrl)
            .allowHardware(enableHardware)
            .crossfade(true)
            .size(avatarSize) // 限制本地加载尺寸
            .scale(Scale.FILL)
            .memoryCachePolicy(CachePolicy.ENABLED)
            .diskCachePolicy(CachePolicy.ENABLED)
            
        // 为相关域名添加Referer头
        if (url.contains("chat-img.jwznb.com") || url.contains("jwznb.com") || url.contains("myapp.jwznb.com")) {
            builder.setHeader("Referer", "https://myapp.jwznb.com")
            builder.setHeader("User-Agent", "Mozilla/5.0 (Linux; Android 10; Mobile) AppleWebKit/537.36")
        }
        
        return builder.build()
    }
    
    /**
     * 创建机器人图片请求 - 内存优化版
     */
    fun createBotImageRequest(
        context: Context,
        url: String,
        enableHardware: Boolean = true,
        size: Int = ImageSize.AVATAR_LARGE
    ): ImageRequest {
        if (isDataSaverEnabled(context)) {
            return ImageRequest.Builder(context)
                .data(null)
                .build()
        }
        return ImageRequest.Builder(context)
            .data(url)
            .setHeader("Referer", "https://myapp.jwznb.com")
            .setHeader("User-Agent", "Mozilla/5.0 (Linux; Android 10; Mobile) AppleWebKit/537.36")
            .allowHardware(enableHardware)
            .crossfade(true)
            .size(size)
            .scale(Scale.FIT)
            .memoryCachePolicy(CachePolicy.ENABLED)
            .diskCachePolicy(CachePolicy.ENABLED)
            .build()
    }
    
    /**
     * 创建分区相关图片请求 - 内存优化版
     */
    fun createBoardImageRequest(
        context: Context,
        url: String,
        enableHardware: Boolean = true,
        maxSize: Int = ImageSize.THUMBNAIL
    ): ImageRequest {
        if (isDataSaverEnabled(context)) {
            return ImageRequest.Builder(context)
                .data(null)
                .build()
        }
        return ImageRequest.Builder(context)
            .data(url)
            .setHeader("Referer", "https://myapp.jwznb.com")
            .setHeader("User-Agent", "Mozilla/5.0 (Linux; Android 10; Mobile) AppleWebKit/537.36")
            .allowHardware(enableHardware)
            .crossfade(true)
            .size(maxSize)
            .scale(Scale.FIT)
            .memoryCachePolicy(CachePolicy.ENABLED)
            .diskCachePolicy(CachePolicy.ENABLED)
            .build()
    }
    
    /**
     * 创建大图预览请求 - 用于ImageViewer
     * 先加载缩略图，再加载原图，减少内存峰值
     * @param forThumbnail true表示加载缩略图，false表示加载原图
     */
    fun createPreviewImageRequest(
        context: Context,
        url: String,
        forThumbnail: Boolean = true
    ): ImageRequest {
        val builder = ImageRequest.Builder(context)
            .data(url)
            .crossfade(true)
            .memoryCachePolicy(CachePolicy.ENABLED)
            .diskCachePolicy(CachePolicy.ENABLED)
        
        if (forThumbnail) {
            // 缩略图模式: 限制尺寸，快速加载
            builder.size(ImageSize.PREVIEW)
            builder.scale(Scale.FIT)
            builder.allowHardware(true)
        } else {
            // 原图模式: 不限制尺寸，但禁用硬件加速以支持缩放
            builder.size(Size.ORIGINAL)
            builder.allowHardware(false)
        }
            
        // 为相关域名添加Referer头
        if (url.contains("chat-img.jwznb.com") || url.contains("jwznb.com") || url.contains("myapp.jwznb.com")) {
            builder.setHeader("Referer", "https://myapp.jwznb.com")
            builder.setHeader("User-Agent", "Mozilla/5.0 (Linux; Android 10; Mobile) AppleWebKit/537.36")
        }
        
        return builder.build()
    }
}
